### DISCUSSION TITLE
Changing table width...

### DISCUSSION THREAD

**Discussion by jodros** [Omikhleia/resilient.sile#186] [Q&A]
I'm creating a table in the markdown way, and didn't find any case among the examples where their width is defined in `styles.yml` and in the docs the example also doesn't mention  how to define a specific width. Again I don't want to use that verbose and tiresome SIL ptable syntax...

In this case, there's too much void space in the first column, and should decrease its width while increasing it for the other side:

![изображение](https://github.com/user-attachments/assets/a29be4a4-ba65-4eba-8235-6d01a2683504)

P.S.: I'm also curious to know whether it's possible to change their colors from the yml...

**Comment by Omikhleia** [comment]
If it's not in the documentation... well, it might be that it doesn't exist yet...

Table styling (a long standing issue https://github.com/Omikhleia/ptable.sile/issues/6) - not to say table configuration from Djot/Markdown[^1] - isn't addressed at this day.

The ptable package is one of the first thing I did back in late 2021 -- SILE had no decent table support back then -- and predates the styling paradigm introduced in "omibook" (later re-invented as "resilient") by several months.
I never found the time to reconcile these, and the fact that ptable lives as its own independent package (that [some people](https://github.com/sile-typesetter/casile/commit/0c3ebb7cefa7927b9138930951ced992f7a2b610) are finally using, outside of resilient) doesn't make it easier.
I'll be honest here, my own priorities are elsewhere:
 - Having bibliographies = This summer work on SILE should eventually see the light, but then there's a lot of work to be done on resilient.sile and markdown.sile to use it there...
 - Having better math support = I'm currently on it, as might have been noticed...
 - Having a style-enabled and more powerful indexer package (https://github.com/Omikhleia/resilient.sile/issues/17, referring also to https://github.com/sile-typesetter/sile/issues/1339)

Contributions would be welcome, but nothing is simple here, in any of these areas.

[^1]: Likely Djot only. We can't have attributes on tables in Markdown...

**Reply by Omikhleia** [reply]
> ... where their width is defined in styles.yml and in the docs the example also doesn't mention how to define a specific width.

On a second thought here:
 - It can't really be a style, each table may have different contraints depending on the content.
 - If we were considering Djot rather that Markdown, we could support some explicit table attributes...
    - This is what witiko/markdown does, IIRC, but with a (custom) Markdown extension....

EDIT: So it might be a question/issue for markdown.sile, if we eventually consider the second option.

EDIT2: or ptable.sile ought to be more clever, and perform the real (e.g. CSS table layout) algorithm, computing all minimal widths first and applying a heuristics for auto-adjustment. It does cost an extra pass on the cells, however. IIRC.

EDIT3: Of course, all solutions could make sense, so in the end, with some option for selection...
  - No attempt at auto-adjustment = as of now, fast but not always satisfying
  - Explicit user-chosen widths via attributes (in Djot or SIL) = fast too, solves the problem "manually"
  - Clever auto-adjustment logic in ptable (based on the CSS algorithm or not) = extra perf. cost, but often optimal without tinkering
